scripts/gen_pb_options.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python3
+
+import argparse
+import re
+import sys
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("file", help="input file")
+parser.add_argument("-o", dest="output_file", help="output file")
+
+args = parser.parse_args()
+
+with open(args.file, encoding="utf-8") as file:
+    input_lines = [line.rstrip() for line in file]
+
+with open(args.output_file, "w", encoding="utf-8") if args.output_file else sys.stdout as file:
+    file.write(
+        """#include <iostream>
+#include <fstream>
+#include <regex>
+#include <string>
+
+"""
+    )
+
+    # write required headers
+
+    file.writelines([line + "\n" for line in input_lines if line.strip().startswith("#include")])
+    file.write("\n")
+
+    # GetVarValue function
+
+    file.write("static size_t GetVarValue(const std::string& name)\n{")
+
+    for line in input_lines:
+        # remove comments
+        line = line.split("#")[0].split("//")[0]
+        if len(line) == 0:
+            continue
+
+        for config in re.findall(r"\${([^}]+)}", line):
+            file.write('    if (name == "{}") {{\n'.format(config))
```

Comment:
double {,  probably a mistake
and you can use formatted literal strings f" if (name == "{config}"